dolt_integrations/dvc/examples/fashion-mnist/dolt-dvc-demo.md

```diff
@@ -0,0 +1,610 @@
++-----------------------+------------+----------------------------------+
+| 0.6217                | 10000      | vq01vp8o1fsdq95eo4sutmppfhovejfv |
+| 0.5352                | 10000      | foo013e1hgej8oa0282bv1ecb9280mlh |
++-----------------------+------------+----------------------------------+
+```
+
+We use the `dolt_history` table above to access every version of
+the prediction table, compare results, and calculate overall accuracy.
+This query can be broken down into 1) a `with` block that adds a correct
+column that's either 0 or 1, and 2) a `select` statement that performs
+accuracy arithmetic, and 3) a `group by` to bucket accuracy by training
+run.
+Computing every training accuracy requires the history table because predictions are
+replaced every iteration.
+
+We also added a summary row each training run, an easier way to
+incrementally generate logs:
+```
+> > dolt sql -q "select * from summary"
++-----------+--------------+--------------------------------------+--------+
+| loss      | label_branch | timestamp                            | acc    |
++-----------+--------------+--------------------------------------+--------+
+| 1.6994121 | fashion      | 2021-04-02 10:29:12.840018 +0000 UTC | 0.5352 |
+| 1.580542  | master       | 2021-04-02 10:43:14.570946 +0000 UTC | 0.6217 |
++-----------+--------------+--------------------------------------+--------+
+```
+
+One instance where we might want to access the commit history is
+`dolt blame`’ing a data changes. For example, identifying who added
+certain rows to the fashion branch:
+```bash
+dolt blame fashion labels | head -n 5
++--------+----------------------------+-----------------+------------------------------+----------------------------------+
+| ROW_ID | COMMIT MSG                 | AUTHOR          | TIME                         | COMMIT                           |
++--------+----------------------------+-----------------+------------------------------+----------------------------------+
+| 14648  | Add Fashion labels         | Bojack Horseman | Wed Mar 31 16:20:50 PDT 2021 | 5fblpjp5neurvsfp989s6ea9lt01vd2q |
+| 27051  | Add Fashion labels         | Bojack Horseman | Wed Mar 31 16:20:50 PDT 2021 | 5fblpjp5neurvsfp989s6ea9lt01vd2q |
+```
+
+You can learn more about how to use Dolt’s Git and SQL features
+[here](https://docs.dolthub.com).
+
+## How Dolt Integrates With DVC
+
+### Adding
+
+DVC’s main metadata is the md5-hash used to sync local and remote
+filesystems. Usually, DVC hashes files and directories the same
+way as Git. To avoid duplicating what Dolt does internally, we
+substituted Dolt’s commits in-lieu of an md5 hash.
+
+```bash
+> cat data/labels.dvc
+outs:
+- md5: tkcao72e0upe2umb2kokartgivd9keqc-pinisl8m4tfdecb8iikqhicta61nou9s.dolt
+  size: 1465033
+  path: labels
+  cache: false
+```
+
+Dolt's three  heads tell us 1) the last commit, 2) the hash if
+all files in our working directory were committed now
+, 3) the ongoing index/staging commit hash.
```

Comment:
Dolt doesn't track files though? I assume you mean DVC?

Reply:
right, rows not files